src/main/java/teammates/ui/newcontroller/ConfirmFeedbackSessionSubmissionAction.java

```
@@ -0,0 +1,115 @@
+                        courseId, feedbackSessionName, studentAttributes.getEmail(), false, true);
+            }
+            if (isSubmissionEmailConfirmationEmailRequested) {
+                email = new EmailGenerator().generateFeedbackSubmissionConfirmationEmailForStudent(
+                            feedbackSession, studentAttributes, Instant.now());
+            }
+            break;
+        case INSTRUCTOR_SUBMISSION:
+            InstructorAttributes instructorAttributes = getInstructorOfCourseFromRequest(feedbackSession.getCourseId());
+            boolean hasInstructorRespondedForSession =
+                    logic.hasGiverRespondedForSession(instructorAttributes.getEmail(), feedbackSessionName, courseId);
+            if (hasInstructorRespondedForSession) {
+                taskQueuer.scheduleUpdateRespondentForSession(
+                        courseId, feedbackSessionName, instructorAttributes.getEmail(), true, false);
+            } else {
+                taskQueuer.scheduleUpdateRespondentForSession(
+                        courseId, feedbackSessionName, instructorAttributes.getEmail(), true, true);
+            }
+            if (isSubmissionEmailConfirmationEmailRequested) {
+                email = new EmailGenerator().generateFeedbackSubmissionConfirmationEmailForInstructor(
+                        feedbackSession, instructorAttributes, Instant.now());
+            }
+            break;
+        default:
+            throw new InvalidHttpParameterException("Unknown intent " + intent);
+        }
+
+        if (email != null) {
+            try {
+                emailSender.sendEmail(email);
+            } catch (EmailSendingException e) {
+                log.severe("Submission confirmation email failed to send: "
+                        + TeammatesException.toStringWithStackTrace(e));
+                return new JsonResult("Submission confirmation email failed to send",
+                        HttpStatus.SC_INTERNAL_SERVER_ERROR);
```

I don't think you should return an error here? A failed email, while not desirable, does not mean a failed submission.